app/gui/src/controller/searcher.rs

```diff
@@ -322,11 +322,7 @@ impl ThisNode {
     ///
     /// Returns `None` if the given node's information cannot be retrieved or if the node does not
     /// introduce a variable.
-    pub fn new(
-        nodes: Vec<double_representation::node::Id>,
-        graph: &controller::Graph,
-    ) -> Option<Self> {
-        let id = *nodes.first()?;
+    pub fn new(id: double_representation::node::Id, graph: &controller::Graph) -> Option<Self> {
```

Comment:
Passing `selected_nodes` everywhere just to take the first of it doesn't make sense to me. So now it receives just one node, either selected one or the source node of the connection (depending on Mode)

Reply:
I think there was an idea long ago, that the new node in searcher shall take not only "this"/"self" argument of selected node, but also arguments from other selected nodes. But for now, we can leave your fix, according to the YAGNI rule.